pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,270 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	runShared "github.com/cli/cli/pkg/cmd/run/shared"
+	"github.com/cli/cli/pkg/cmd/workflow/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/markdown"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    cmdutil.Browser
+
+	Selector string
+	Ref      string
+	Web      bool
+	Prompt   bool
+	Raw      bool
+	Yaml     bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := &ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Browser:    f.Browser,
+	}
+
+	cmd := &cobra.Command{
+		Use:    "view [<run-id> | <file name>]",
+		Short:  "View the summary of a workflow",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		Example: heredoc.Doc(`
+		  # Interactively select a workflow to view
+		  $ gh workflow view
+
+		  # View a specific workflow
+		  $ gh workflow view 0451
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.Raw = !opts.IO.CanPrompt()
```

Comment:
I'm not sure whether `CanPrompt` is the accurate predicate here. We are asking whether stdout is a TTY, right? There is `opts.IO.IsStdoutTTY()`